core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -346,7 +346,7 @@ object Pull extends PullLowPriority {
      */
     private[fs2] def uncons: Pull[F, Nothing, Option[(Chunk[O], Pull[F, O, Unit])]] =
       self match {
-        case Succeeded(_)    => Succeeded(None)
+        case _: Succeeded[_] => Succeeded(None)
```

Comment:
Before:

```java
  public final <F, O> Pull<F, Nothing$, Option<Tuple2<Chunk<O>, Pull<F, O, BoxedUnit>>>> uncons$extension(Pull<F, O, BoxedUnit> $this) {
      Pull<F, O, BoxedUnit> pull = $this;
      if (pull instanceof Pull.Succeeded) {
          return new Pull.Succeeded<None$>(None$.MODULE$);
      }
      if (pull instanceof Pull.Output) {
          Pull.Output output = (Pull.Output)pull;
          Chunk vals = output.values();
          return new Pull.Succeeded<Some>(new Some((Object)Predef.ArrowAssoc$.MODULE$.$minus$greater$extension(Predef$.MODULE$.ArrowAssoc(vals), Pull$.fs2$Pull$$unit)));
      }
      if (pull instanceof Pull.Fail) {
          Pull.Fail fail = (Pull.Fail)pull;
          return fail;
      }
      if (pull instanceof Pull.Interrupted) {
          Pull.Interrupted interrupted = (Pull.Interrupted)pull;
          return interrupted;
      }
      return new Pull.Uncons<F, O>($this);
  }
```

After:
```java
  public final <F, O> Pull<F, Nothing$, Option<Tuple2<Chunk<O>, Pull<F, O, BoxedUnit>>>> uncons$extension(Pull<F, O, BoxedUnit> $this) {
      Pull<F, O, BoxedUnit> pull = $this;
      if (pull instanceof Pull.Succeeded) {
          return new Pull.Succeeded<None$>(None$.MODULE$);
      }
      if (pull instanceof Pull.Output) {
          Pull.Output output = (Pull.Output)pull;
          Chunk vals = output.values();
          return new Pull.Succeeded<Some>(new Some((Object)Predef.ArrowAssoc$.MODULE$.$minus$greater$extension(Predef$.MODULE$.ArrowAssoc(vals), Pull$.fs2$Pull$$unit)));
      }
      if (pull instanceof Pull.Fail) {
          Pull.Fail fail = (Pull.Fail)pull;
          return fail;
      }
      if (pull instanceof Pull.Interrupted) {
          Pull.Interrupted interrupted = (Pull.Interrupted)pull;
          return interrupted;
      }
      return new Pull.Uncons<F, O>($this);
  }
```

Reply:
Looks like the same code, good to know, thank you! :)